### DISCUSSION TITLE
Frontend / Backend

### DISCUSSION THREAD

**Discussion by fdecle** [Viladoman/CompileScore#48] [Q&A]
Hi,

I didn't find a doc that explain what are exactly the frontend/backend steps... More generally, it would be great to have some tips about how to interpret the results, what you should look after when you have a long compilation time... is there such a doc somewhere ?

**Comment by Viladoman** [comment]
I haven't done anything like that yet. I agree that having some guidelines on understanding the information exposed by the compiler and actions to take depending on what Compile Score reports is a really good idea. 

I will try gathering data and examples and start a wiki page for this. Thanks for the suggestion. 

Answering your question: 
- FrontEnd would be the compiler 'understanding the code': (Parsing code, creating the class memory layours, instancing templates, generating an AST ) 
- BackEnd would grab the output from the FrontEnd and generate the object file/binary ( generate target code, optimize code .... ) 

In my experience most large codebases suffer from massive badly done includes.
